railties/lib/rails/generators/rails/app/templates/config/puma.rb

```diff
@@ -0,0 +1,9 @@
+threads_count = ENV.fetch("MAX_THREADS") { 5 }.to_i
+threads threads_count, threads_count
+
+port        ENV.fetch("PORT") { 3000 }
+environment ENV.fetch("RAILS_ENV") { "development" }
+
+on_worker_boot do
```

Comment:
I'm not quite sure where it should live / look like, but I think we should probably provide a method somewhere that does this... allowing components to separately register such behaviour.